src/hb-cff-interp-dict-common.hh

```diff
@@ -125,130 +95,67 @@ struct dict_opset_t : opset_t<number_t>
     }
   }
 
+  /* Turns CFF's BCD format into strtod understandable string */
   static double parse_bcd (byte_str_ref_t& str_ref)
   {
-    bool    neg = false;
-    double  int_part = 0;
-    uint64_t frac_part = 0;
-    uint32_t  frac_count = 0;
-    bool    exp_neg = false;
-    uint32_t  exp_part = 0;
-    bool    exp_overflow = false;
-    enum Part { INT_PART=0, FRAC_PART, EXP_PART } part = INT_PART;
+    hb_vector_t<char> buf;
```

Comment:
Isn't the maximum length capped?  Just allocate that much on stack I suggest.

Reply:
Done on #1955